rfqservice/negotiator.go

```diff
@@ -0,0 +1,279 @@
+package rfqservice
+
+import (
+	"encoding/hex"
+	"fmt"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/rfqmsg"
+)
+
+// NegotiatorCfg holds the configuration for the negotiator.
+type NegotiatorCfg struct {
+	// PriceOracle is the price oracle that the negotiator will use to
+	// determine whether a quote is accepted or rejected.
+	PriceOracle PriceOracle
+}
+
+// Negotiator is a struct that handles the negotiation of quotes. It is a RFQ
+// subsystem. It determines whether a quote is accepted or rejected.
+type Negotiator struct {
+	startOnce sync.Once
+	stopOnce  sync.Once
+
+	// cfg holds the configuration parameters for the negotiator.
+	cfg NegotiatorCfg
+
+	// outgoingMessages is a channel which is populated with outgoing peer
+	// messages.
+	outgoingMessages chan<- rfqmsg.OutgoingMsg
+
+	// assetSellOffers is a map (keyed on asset) that holds the asset sell
+	// offers that the negotiator has received.
+	assetSellOffers map[string]AssetSellOffer
+
+	assetGroupSellOffers map[string]AssetSellOffer
```

Comment:
Any particular reason for a `string` over the asset ID byte array?

Reply:
Latest commits use:
```
	assetSellOffers map[asset.ID]SellOffer

	// assetGroupSellOffers is a map (keyed on asset group key) that holds
	// asset sell offers.
	assetGroupSellOffers map[btcec.PublicKey]SellOffer
```